.ci/generate_test_report_lib_test.py

```diff
@@ -246,6 +288,58 @@ def test_no_failures_build_failed(self):
             ),
         )
 
+    def test_no_failures_build_failed_ninja_log(self):
+        self.assertEqual(
+            generate_test_report_lib.generate_report(
+                "Foo",
+                1,
+                [
+                    junit_from_xml(
+                        dedent(
+                            """\
+          <?xml version="1.0" encoding="UTF-8"?>
+          <testsuites time="0.00">
+          <testsuite name="Passed" tests="1" failures="0" skipped="0" time="0.00">
+          <testcase classname="Bar/test_1" name="test_1" time="0.00"/>
+          </testsuite>
+          </testsuites>"""
+                        )
+                    )
+                ],
+                [
+                    [
+                        "[1/5] test/1.stamp",
+                        "[2/5] test/2.stamp",
+                        "[3/5] test/3.stamp",
+                        "[4/5] test/4.stamp",
+                        "FAILED: test/4.stamp",
+                        "touch test/4.stamp",
+                        "Wow! Close To You!",
+                        "[5/5] test/5.stamp",
+                    ]
+                ],
+            ),
+            (
+                dedent(
+                    """\
+                    # Foo
+
+                    * 1 test passed
```

Comment:
If there are test and build failures, should be label the build parts?

Especially in this case. I see that 1 test passed and there is this details thing that I won't know was a failure until I click into it.

Istr an "all tests passed but the build failed" message somewhere, that would work here too.

And I know we're rapidly piling on combinations here, not easy to make sense of every one.

Reply:
If the number of circumstances to handle becomes an issue, we could consider a more rigid format where there's always a build and test section but they just say "passed" if nothing unusual happened. Instead of building this semi-dynamic thing which is nice, but might get out of hand.

Not something for this set of changes but if you feel like it'd save you a headache, it's one option.

Reply:
We can't trivially just report build failures along with test failures because the `check-*` targets will show up as build failures given they are run through ninja. We should be able to detect that and filter them out, but I think it's reasonable to leave them out of scope for now.

Reply:
Sorry, I think I missed what you meant in your original comment. I thought you were talking about the case where there are test failures *and* other targets that fail to compile/link. That gets difficult for the reason I mentioned.

Within this test we handle the case where there are no test failures (but some ran), but there are build failures. I added some text (similar to the other case) to check the failures.